packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -542,30 +540,37 @@ class HotRunner extends ResidentRunner {
     // Check if the isolate is paused and resume it.
     final List<Future<void>> operations = <Future<void>>[];
     for (final FlutterDevice device in flutterDevices) {
-      final Set<Isolate> uiIsolates = <Isolate>{};
+      final Set<String> uiIsolatesIds = <String>{};
       for (final FlutterView view in device.views) {
         if (view.uiIsolate == null) {
           continue;
         }
-        uiIsolates.add(view.uiIsolate);
+        uiIsolatesIds.add(view.uiIsolate.id);
         // Reload the isolate.
-        operations.add(view.uiIsolate.reload().then((ServiceObject _) {
-          final ServiceEvent pauseEvent = view.uiIsolate.pauseEvent;
-          if ((pauseEvent != null) && pauseEvent.isPauseEvent) {
+        final Future<vm_service.Isolate> reloadIsolate = device.vmService
+          .getIsolate(view.uiIsolate.id)
+          .catchError((dynamic error, StackTrace stackTrace) {
+            return null;
+          }, test: (dynamic error) => error is vm_service.SentinelException);
+        operations.add(reloadIsolate.then((vm_service.Isolate isolate) async {
+          if ((isolate != null) && isPauseEvent(isolate.pauseEvent.kind)) {
             // Resume the isolate so that it can be killed by the embedder.
-            return device.vmService.resume(view.uiIsolate.id);
+            await device.vmService.resume(view.uiIsolate.id);
           }
-          return null;
         }));
       }
+
       // The engine handles killing and recreating isolates that it has spawned
       // ("uiIsolates"). The isolates that were spawned from these uiIsolates
       // will not be restared, and so they must be manually killed.
-      for (final Isolate isolate in device?.flutterDeprecatedVmService?.vm?.isolates ?? <Isolate>[]) {
-        if (!uiIsolates.contains(isolate)) {
-          operations.add(isolate.invokeRpcRaw('kill', params: <String, dynamic>{
-            'isolateId': isolate.id,
-          }));
+      final vm_service.VM vm = await device.vmService.getVM();
+      for (final vm_service.IsolateRef isolateRef in vm.isolates) {
+        if (!uiIsolatesIds.contains(isolateRef.id)) {
```

Comment:
Consider flipping the sense of the `if` with a `continue` under it, and unindenting the rest of the loop body.

Reply:
Done